### PR TITLE
Copy Action workflow files from module template

### PR DESCRIPTION
**Pull request by mcmire**
This PR aligns the workflow files with our module template so we have less to think about when maintaining this library. It also adds the ability to create a release PR so that we can issue releases more easily.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,6 +39,8 @@
   },
   "homepage": "https://github.com/MetaMask/detect-provider#readme",
   "devDependencies": {
+    "@lavamoat/allow-scripts": "^2.0.3",
+    "@metamask/auto-changelog": "^2.5.0",
```
Comment (rekmarks) [inline package.json:43]: There's a more recent version.

**Review by rekmarks — COMMENTED** [review]
Can you add a description? You're just replacing the existing workflow files with more up-to-date ones?

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by meny3437-blip — COMMENTED** [review]
npm i @metamask/detect-provider meny3437-blip

**Pull request merged**